libcrux-ml-kem/src/vector/portable/ntt.rs

```diff
@@ -138,17 +222,64 @@ pub(crate) fn ntt_multiply_binomials(
     j: usize,
     out: &mut PortableVector,
 ) {
-    let o0 = montgomery_reduce_element(
-        (a.elements[i] as i32) * (b.elements[i] as i32)
-            + (montgomery_reduce_element((a.elements[j] as i32) * (b.elements[j] as i32)) as i32)
-                * (zeta as i32),
-    );
-    let o1 = montgomery_reduce_element(
-        (a.elements[i] as i32) * (b.elements[j] as i32)
-            + (a.elements[j] as i32) * (b.elements[i] as i32),
-    );
+    let ai = a.elements[i];
+    let bi = b.elements[i];
+    let aj = a.elements[j];
+    let bj = b.elements[j];
+    hax_lib::fstar!("assert(Spec.Utils.is_i16b 3328 $ai);
+                     assert(Spec.Utils.is_i16b 3328 $bi);
+                     assert(Spec.Utils.is_i16b 3328 $aj);
+                     assert(Spec.Utils.is_i16b 3328 $bj);
+                     assert_norm (3328 * 3328 < pow2 31)");         
+            
+    hax_lib::fstar!("Spec.Utils.lemma_mul_i16b 3328 3328 $ai $bi");
+    let ai_bi = (ai as i32) * (bi as i32);
```

Comment:
Please use variable names that are a telling the reader what's going on. Also, the order of operations seems a little odd. `ai_bi` isn't used until l243 for example.

Reply:
The proof here is very heavy and fragile to small changes. `ai_bi` can probably be moved but not a clear benefit to do so?

Reply:
As you can see in the calc proof below, we use these variable names quite a lot, and longer and more descriptive names will make the F* proof much more verbose. Can be done, of course, if you have a suggestion for how to name `ai * bi * zeta` etc.

Reply:
Yes, but verbose is what we're going for. Verbosity is there to make it understandable. I think it's ok in this function. But this is a general issue I'll have an eye on more in future.